sycl/include/CL/sycl/detail/kernel_program_cache.hpp

```diff
@@ -25,29 +25,35 @@ namespace detail {
 class context_impl;
 class KernelProgramCache {
 public:
+  struct BuildResultT {
+    std::string Msg;
+    cl_int Code;
+  };
+
   /// Denotes pointer to some entity with its state.
   /// The pointer is not null if and only if the entity is usable.
   /// State of the entity is provided by the user of cache instance.
   /// Currently there is only a single user - ProgramManager class.
   template<typename T>
-  struct EntityWithState {
+  struct EntityWithBuildResult {
     std::atomic<T *> Ptr;
     std::atomic<int> State;
+    std::unique_ptr<BuildResultT> BuildResult;
```

Comment:
Why this is a pointer? If it is just to test for an error, then the error code or the status should be enough.

Reply:
Done

Reply:
Error code can be zero (which is a bug but not a subject of this PR). Thus added a distinct bool here.

Reply:
Then why `BS_Failed` is not enough?

Reply:
Then it would be nice to have a comment for FilledIn field. Otherwise someone later may try to remove it as duplicate.

Reply:
Done